src/template.js

```diff
@@ -35,12 +44,37 @@ switch (data.method) {
       break;
     }
 
-    if (queryPermission('inject_script', INSIGHTS_LIBRARY_URL)) {
+    const url = getLibraryURL(data.useIIFE);
+
+    if (queryPermission('inject_script', url)) {
       injectScript(
-        INSIGHTS_LIBRARY_URL,
-        data.gtmOnSuccess,
+        url,
+        () => {
+          if (!copyFromWindow('aa')) {
+            data.gtmOnFailure();
+            logger('[ERROR] Failed to load search-insights.');
+            return;
+          }
+
+          let libraryLoaded = false;
+          aa('getUserToken', null, () => {
+            // call any method to see if it reacts.
+            libraryLoaded = true;
+          });
+          callLater(() => {
```

Comment:
I'm not sure I understand why this needs to be wrapped in a callLater (nor what callLater is supposed to do). Do you think you could add a comment to explain the reasons behind it?

Reply:
You're right. It was my bad. It's removed now. https://github.com/algolia/search-insights-gtm/pull/12/commits/8b2a55ba9cbb0ce9d5683f4b22bc1bc6582324df